src/preset.ts

```diff
@@ -988,12 +997,129 @@ async function expandConfigurePresetImpl(folder: string, name: string, workspace
     return null;
 }
 
-async function tryApplyVsDevEnv(preset: ConfigurePreset) {
-    if (!preset.__vsDevEnvApplied) {
-        let compilerEnv = EnvironmentUtils.createPreserveNull();
-        // [Windows Only] If CMAKE_CXX_COMPILER or CMAKE_C_COMPILER is set as cl, clang, clang-cl, clang-cpp and clang++,
-        // but they are not on PATH, then set the env automatically.
-        if (process.platform === 'win32') {
+export interface VsDevEnvOptions {
+    preset: ConfigurePreset;
+    shouldInterrogateForNinja: boolean;
+    compilerName?: string; // Only will have a value when `useVsDeveloperEnvironmentMode` is "auto"
+}
+
+/**
+ * @param opts Options to control the behavior of obtaining the VS developer environment.
+ * @returns Either the VS developer environment or undefined if it could not be obtained.
+ */
+async function getVsDevEnv(opts: VsDevEnvOptions): Promise<EnvironmentWithNull | undefined> {
+    const arch = getArchitecture(opts.preset);
```

Comment:
Our presets do not specify the architecture or toolset architecture by design, so that they can be used for native compilation on both x64 and arm64 machines. `getArchitecture` and `getToolset` return `x86` as a default value, which we don't support as a compilation target. Could they be changed to return a value based on the processor architecture? I believe this is consistent with the vcvarsall behavior when host and target arch are not specified

Reply:
Would something like using the `util.getHostArchitecture` method do what you're describing and get what you're expecting? We could default to that method rather than simply `x86`.

Reply:
Updated. 👍